pkg/controller/restore_controller.go

```diff
@@ -41,8 +41,13 @@ import (
 	listers "github.com/heptio/ark/pkg/generated/listers/ark/v1"
 	"github.com/heptio/ark/pkg/restore"
 	"github.com/heptio/ark/pkg/util/collections"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+// NonRestorableResources is a blacklist for the restoration process. Any resources
```

Comment:
Leading lowercase n

Reply:
@jrnt30 bump

Reply:
I'll try and take a look on the train on that work.

Reply:
Looking at this closer again, I suppose I related the blacklist of resources similarly to the setting of a default as we do with the `IncludedResources` and `IncludedNamespaces`.  

If we push this down to the `pkg/restore` I don't think as cleanly segregates the validation logic being exposed to the client either.  

Were you thinking that we put this in the `kubernetesRestorer.Restore` directly or simply a utility function checking these things?  I'm afraid I may be missing part of your thought process here (it may be simply that you are proposing a larger refactor than I am considering looking at the type signature here).

Reply:
Let's say that someone decides to pull in `github.com/heptio/ark/pkg/...` to use Ark as a library (we already know of at least one project interested in doing so). If the logic to exclude nodes is only in the controller, then anyone using the library directly and not using the controller would end up restoring nodes unless they also write logic to exclude them.

The more I think about this, given that the other defaulting logic is in the controllers, I think it's fine to include this here for now in the controller. We may decide to move more logic "down" in the future, but I don't think that possibility needs to hold up this PR.

Reply:
Understood, thanks.  I do think having that more centralized may be beneficial in those cases. 

Out of curiosity, is the project you are referring to public?

Reply:
https://github.com/kris-nova/kubicorn/issues/195